conda/recipes/pynvjitlink/meta.yaml

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) 2018-2024, NVIDIA CORPORATION.
+
+{% set data = load_file_data("pyproject.toml") %}
+
+package:
+  name: pynvjitlink
+  version: {{ data.get("project", {}).get("version", "") }}
+
+source:
+  path: ../../..
+
+build:
+  script:
+    - {{ PYTHON }} -m pip install . -vv
+  ignore_run_exports_from:
+    - libnvjitlink-dev
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}
+    - cuda-version {{ cuda_compiler_version }}
+    - cmake >=3.26.4
+    - ninja
+    - sysroot_{{ target_platform }} 2.17
+  host:
+    - libnvjitlink-dev
+    - libnvjitlink-static
+    - cuda-version {{ cuda_compiler_version }}
+    - python
+    - pip
+    - scikit-build-core
+  run:
+    - python
+    - numba >=0.57
+    - {{ pin_compatible('cuda-version', max_pin='x', min_pin='x') }}
```

Comment:
```suggestion
    - {{ pin_compatible('cuda-version', min_pin='x', max_pin=(cuda_major ~ "." ~ (cuda_minor + 1))) }}
```

Reply:
Is this valid? I thought `pin_compatible` only supported pinning expressions (i.e. literal strings of the form `x[.x[.x...]]`, not actual versions. Maybe we can use the `upper_bound` parameter for what you're trying to do here? https://docs.conda.io/projects/conda-build/en/stable/resources/variants.html#extra-jinja2-functions

Reply:
Yes, I think we want `upper_bound`. Fixed in 108e2bf0b7bd241316cce69840a732521554f4d9.